ext/json/json.c

```diff
@@ -178,6 +178,25 @@ PHP_JSON_API zend_result php_json_decode_ex(zval *return_value, const char *str,
 }
 /* }}} */
 
+/* {{{ */
+PHP_JSON_API zend_result php_json_validate_ex(zval *return_value, const char *str, size_t str_len, zend_long options, zend_long depth)
+{
+	php_json_parser parser;
+
+	const php_json_parser_methods parser_validate_methods = get_validate_methods();
+	php_json_parser_init_ex(&parser, return_value, str, str_len, (int)options, (int)depth, &parser_validate_methods);
+
+	if (php_json_yyparse(&parser)) {
+		php_json_error_code error_code = php_json_parser_error_code(&parser);
+		JSON_G(error_code) = error_code;
+		RETVAL_FALSE;
+		return FAILURE;
+	}
+
+	return SUCCESS;
+}
```

Comment:
Am I missing something, but why not just return `bool` and drop the in-out `return_value` param? As you could then just do below:
``RETURN_BOOL(php_json_validate_ex(str, str_len, options, depth));``

Could also argue about using a zend_string to not need to carry the length of the string separately.

Reply:
Let me check this once I get Back Home and I Will provide you an answer. Need to check the code and notes

Reply:
- Returning bool instead and removing return_value sounds reasonable
- I think it's similar to the reason why `php_json_parser_init_ex` and other `PHP_JSON_API` (externally visible symbols) - so that **other** PECLs have the ability to check if json is valid, without needing to waste time/memory to copy a C string into a zend_string to allocate and release it

Reply:
I am back, had to re-install all my laptop OS, software, etc.

1) Removing `return_value`

The reason I am using it, is because I was following the code structure used for json_decode().

I tried to add my change in a way that the overall file looks like it was written by the same person; also I try not to change too much.

Plus, I need ``return_value in line 187 as a second parameter:
```C
php_json_parser_init_ex(&parser, return_value, str, str_len, (int)options, (int)depth, &parser_validate_methods);
```
Please let me know if I am also missing something here. I admit all this is new to me.

2) Returning BOOL instead of FAILURE/SUCCESS

This is something I can try, and I will ... but will only affect this piece of code in json.c

![image](https://user-images.githubusercontent.com/25756860/192045043-d35ec874-833a-4373-a447-7f01db3b2cd5.png)

for something like:
```C
RETURN_BOOL(php_json_validate_ex(return_value, str, str_len, options, depth));
```

Also ... Because we use `return_value` .... **I guess** I could also just have:
`php_json_validate_ex(return_value, str, str_len, options, depth)` 
like with json_decode() , but honestly I dont like how the code looks like in such a way, because at first sight , a new developer like me, does not realize that "the result of this lonely function ... maybe is stored in the parameter return_value"


So I will change locally I will try 2 options:
a. return BOOL instead of SUCCESS/FAILURE
b. just rely on `return_value`

Then I will get back here.

-----------------------------------------------------------------------

Please in the meanwhile let me know if I got your comments right and if you have another though after my comment

Reply:
>The reason I am using it, is because I was following the code structure used for json_decode().

json_decode can return multiple types of zval - json_validate only returns bool

> Plus, I need ``return_value in line 187 as a second parameter:

you can create a variable on the stack `zval tmp;` and pass in `&tmp` - this doesn't allocate anything that needs to be freed when validating.

> `RETURN_BOOL(php_json_validate_ex(return_value, str, str_len, options, depth));`

The `RETURN_TRUE` and `RETURN_BOOL` and `RETURN_*` macros (etc) overwrite the value of return_value immediately so it is just a no-op to use it that way (requiring it but immediately overwrite it in the only caller)

Reply:
Thanks for the hints. I Will prepare the change

Reply:
@Girgias @TysonAndre I did some changes. Hopefuly are in the right direction. 

What Do you think? Thanks in advance